### PR TITLE
FEATURE/7-add-register-endpoint

### PR DESCRIPTION
**Pull request by dmytrovb02**

### CODE REVIEW

File: src/main/java/com/bookstore/dto/user/UserRegistrationRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    private String repeatPassword;
+
+    @NotBlank
+    private String firstName;
+
+    @NotBlank
+    private String lastName;
+
+    private String shippingAddress;
+}
```
Comment (Sanderfuck) [inline src/main/java/com/bookstore/dto/user/UserRegistrationRequestDto.java:41]: Why you don`t validate the shipping address?

File: src/main/java/com/bookstore/model/User.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import jakarta.persistence.Table;
+import lombok.Getter;
+import lombok.Setter;
+
+@Entity
+@Getter
+@Setter
+@Table(name = "users")
+public class User {
+
```
Comment (Sanderfuck) [inline src/main/java/com/bookstore/model/User.java]:
```suggestion
```

**Review by Sanderfuck — CHANGES_REQUESTED** [review]

File: src/main/java/com/bookstore/controller/AuthenticationController.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+@RestController
+@RequestMapping("/api/auth")
+@RequiredArgsConstructor
+public class AuthenticationController {
+    private final UserServiceImpl userService;
+
+    @PostMapping("/register")
+    public UserResponseDto register(@RequestBody @Valid UserRegistrationRequestDto requestDto)
+            throws RegistrationException {
```
Comment (sokimaaa) [inline src/main/java/com/bookstore/controller/AuthenticationController.java]: Never `throws` runtime exception
```suggestion 
{
```

File: src/main/java/com/bookstore/mapper/UserMapper.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package com.bookstore.mapper;
+
+import com.bookstore.config.MapperConfig;
+import com.bookstore.dto.user.UserRegistrationRequestDto;
+import com.bookstore.dto.user.UserResponseDto;
+import com.bookstore.model.User;
+import org.mapstruct.Mapper;
+
+@Mapper(imports = MapperConfig.class, componentModel = "spring")
```
Comment (sokimaaa) [inline src/main/java/com/bookstore/mapper/UserMapper.java]: move `componentModel = "spring"` to MapperConfig

**Pull request merged**